application/controllers/LoginManager.php

```diff
@@ -18,7 +18,92 @@ function force_ssl() {
 	    // }
 	}
 
+	function handleLocalLoginForVueUI() {
+		$requestMethod = $this->input->server('REQUEST_METHOD');
+		if ($requestMethod == 'GET') {
+			return $this->template->publish('vueTemplate');
+		}
+
+		// respond with json
+		header('Content-Type: application/json');
+
+		// if this isn't a POST request, return an error
+		if ($requestMethod != 'POST') {
+			http_response_code(405);
+			echo json_encode([
+				'status' => 'error',
+				'message' => 'invalid request method'
+			]);
+			return;
+		}
+
+		// Get the data from the request body
+		$cleanJSON = $this->security->xss_clean($this->input->raw_input_stream);
+		$data = json_decode($cleanJSON);
+
+		// if the JSON data is invalid, return an error
+		if (!$data) {
+			http_response_code(400);
```

Comment:
We have a helper method for doing this. `render_json` takes a payload and optionally a status code (default is 200)

Reply:
lovely!